src/traces/bar/plot.js

```diff
@@ -479,9 +482,15 @@ function toMoveInsideBar(x0, x1, y0, y1, textBB, opts) {
     }
 
     // compute text and target positions
-    var textX = (textBB.left + textBB.right) / 2;
+    var textX = (
+        (textBB.left + 0 * TEXTPAD) * toLeft +
+        (textBB.right - 0 * TEXTPAD) * toRight
```

Comment:
Do we need to make the `0 * TEXTPAD` explicit? 

I'd prefer

```
        textBB.left * toLeft +
        textBB.right * toRight
```

Reply:
Good eye. 
Thanks. I totally forgot to clean that one.
Fixed by https://github.com/plotly/plotly.js/pull/4469/commits/7098a97097e9e4f2541eb1d2f348ba801b3814a9.

Reply:
Thanks!!